src/MultiFormats.Tests/MultBaseTest.cs

```diff
@@ -0,0 +1,274 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using MultiFormats.Registry;
+
+namespace MultiFormats.Tests
+{
+    [TestClass]
+    public class MultiBaseTest
+    {
+        [TestMethod]
+        public void Codec()
+        {
+            var bytes = new byte[]
+            {
+                1, 2, 3, 4, 5
+            };
+            var bytes1 = MultiBase.Decode(MultiBase.Encode(bytes));
+            var bytes2 = MultiBase.Decode(MultiBase.Encode(bytes, "base16"));
+            CollectionAssert.AreEqual(bytes, bytes1);
+            CollectionAssert.AreEqual(bytes, bytes2);
+        }
+
+        [TestMethod]
+        public void Encode_Unknown_Algorithm()
+        {
+            var bytes = new byte[]
+            {
+                1, 2, 3, 4, 5
+            };
+            ExceptionAssert.Throws<KeyNotFoundException>(() => MultiBase.Encode(bytes, "unknown"));
+        }
+
+        [TestMethod]
+        public void Encode_Null_Data_Not_Allowed()
+        {
+            ExceptionAssert.Throws<ArgumentNullException>(() => MultiBase.Encode(null));
+        }
+
+        [TestMethod]
+        public void Decode_Bad_Formats()
+        {
+            ExceptionAssert.Throws<ArgumentNullException>(() => MultiBase.Decode(null));
+            ExceptionAssert.Throws<ArgumentNullException>(() => MultiBase.Decode(""));
+            ExceptionAssert.Throws<ArgumentNullException>(() => MultiBase.Decode("   "));
+
+            ExceptionAssert.Throws<FormatException>(() => MultiBase.Decode("?"));
+            ExceptionAssert.Throws<FormatException>(() => MultiBase.Decode("??"));
+            ExceptionAssert.Throws<FormatException>(() => MultiBase.Decode("???"));
+            ExceptionAssert.Throws<FormatException>(() => MultiBase.Decode("fXX"));
+        }
+
+        private class TestVector
+        {
+            public string Algorithm { get; set; }
+            public string Input { get; set; }
+            public string Output { get; set; }
+        }
+
+        private TestVector[] TestVectors = new TestVector[]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)